views/js/test/runner/plugins/navigation/next/linearNextItemWarning/test.js

```diff
@@ -0,0 +1,305 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA ;
+ */
+
+define([
+    'core/promise',
+    'taoTests/runner/runner',
+    'taoQtiTest/test/runner/mocks/providerMock',
+    'taoQtiTest/runner/plugins/navigation/next/linearNextItemWarning',
+    'taoQtiTest/runner/helpers/map'
+], function(Promise, runnerFactory, providerMock, pluginFactory, mapHelper) {
+    'use strict';
+
+    var pluginApi;
+    var providerName = 'mock';
+    runnerFactory.registerProvider(providerName, providerMock());
+
+    // mock the item we will get (informational or not)
+    mapHelper.getItemAt = function(map, isInfo) {
```

Comment:
I had difficulties to figure out you were using the `position` property from the `testContext` in order to flag the item as informational. 
I'd suggest to make this part more explicit, for example : 

```js
QUnit.cases([
        {
            title: 'when the next part warning is set',
            context: {
                enableAllowSkipping: false,
                allowSkipping: false,
                options: {
                    nextPartWarning: true,
                    nextSectionWarning: false
                }
            },
            item : {
               informational : false
            }
        },
        {
            title: 'when the item is informational',
            context: {
                isLinear: true,
                options: {
                    nextPartWarning: false,
                    nextSectionWarning: false
                }
            },
            item : {
               informational : false
            }
    ])
    .asyncTest('No dialog is triggered ', function(caseData, assert) {
        var runner = runnerFactory(providerName);
        var plugin = pluginFactory(runner, runner.getAreaBroker());

        // mock test store init
        runner.getTestStore = function() {
            return {
                setVolatile: function() {}
            };
        };
        // mock config
        runner.getTestData = function() {
            return {
                config: caseData.testConfig
            };
        };
        // mock the item we will get (informational or not)
        mapHelper.getItemAt = function(map) {
           return caseData.item;
        };

        QUnit.expect(1);

        //...
    });
```

Reply:
Thanks, I didn't spot that way of doing it. Much clearer now.